testing/go/types_test.go

```diff
@@ -2125,12 +2141,80 @@ var typesTests = []ScriptTest{
 				},
 			},
 			{
-				Query:    `SELECT text 'text' || ' and unknown';`,
-				Expected: []sql.Row{{"text and unknown"}},
+				// Create a unique, secondary index on a TEXT column
+				Query:    "CREATE UNIQUE INDEX v1_unique ON t_text(v1);",
+				Expected: []sql.Row{},
 			},
 			{
-				Query:    `SELECT text 'this is a text string' = text 'this is a text string' AS true;`,
-				Expected: []sql.Row{{"t"}},
+				Query: "SELECT * FROM t_text WHERE v1 = 'World';",
+				Expected: []sql.Row{
+					{2, "World"},
+				},
+			},
+			{
+				// Test the new unique constraint on the TEXT column
+				Query:       "INSERT INTO t_text VALUES (5, 'World');",
+				ExpectedErr: "duplicate unique key given: [World]",
```

Comment:
The error I received from Postgres was:
```
duplicate key value violates unique constraint "v1_unique"
```
Eventually we'll fix our error message such that we're not returning the MySQL variant, and instead returning the same one as Postgres. That would also cause this test to fail. If it were me, I'd just check for the `"unique"` string, since I think it's fairly safe that there's only one logical case where we'd receive a string with such an error in this scenario. That's the approach I try to take, unless the error strings are just wildly different or use completely different words.

This is just something general that I do for testing, to try and make it more robust and reduce false-negatives as we improve our correctness. My other worry is that we tend to take tests as enforcing correctness, so a developer could intentionally prevent a correctness improvement by assuming that the test is the expected output (and not cross-referencing it with a Postgres server, since it's probably assumed it was done with the test initially). I don't think that applies to this particular test, but just something in general with our tests.